twitter-borrowbot/twitterbot.py

```diff
@@ -99,42 +73,106 @@ def edition_unavailable(cls, mention, edition):
 
     @classmethod
     def edition_not_found(cls, mention):
-        print('Replying: Book Not found')
+        # print('Replying: Book Not found')
         cls._tweet(
             mention,
             "sorry, I was unable to spot any books! " +
             "Learn more about how I work here: " +
-            "https://github.com/internetarchive/openlibrary-bots"
+            "https://github.com/internetarchive/openlibrary-bots" +
+            "\nIn short, I need an ISBN10, ISBN13, or Amazon link"
         )
 
+    @classmethod
+    def internal_error(cls, mention):
+        cls._tweet(
+            mention,
+            "Woops, something broke over here! " +
+            "Learn more about how I work here: " +
+            "https://github.com/internetarchive/openlibrary-bots" +
+            "\nIn short, I need an ISBN10, ISBN13, or Amazon Link"
+        )
+
+def get_last_seen_id():
+    with open(STATE_FILE, 'r') as fin:
+        return int(fin.read().strip())
+
+
+def set_last_seen_id(mention):
+    with open(STATE_FILE, 'w') as fout:
+        fout.write(str(mention.id))
+
+def get_parent_tweet_of(mention):
+    return api.get_status(
+        mention.in_reply_to_status_id,
+        tweet_mode="extended")
+
+
+def get_latest_mentions(since=None):
+    since = since or get_last_seen_id()
+    try:
+        return api.mentions_timeline(since, tweet_mode="extended")
+    except Exception as e:
+        print("Exception: %s" % e)
+        return None
+
+def is_reply_to_me(mention):
+    return mention.in_reply_to_status_id is api.me().id
+
+def handle_isbn(mention, isbn):
+    edition = InternetArchive.get_edition(isbn)
+    if edition:
+        if edition.get("availability"):
+            return Tweet.edition_available(mention, edition)
+
+        work = InternetArchive.find_available_work(edition)
+        if work:
+            return Tweet.work_available(mention, work)
+        return Tweet.edition_unavailable(mention, edition)
 
 def reply_to_tweets():
-    mentions = get_latest_mentions()
+    try:
+        mentions = get_latest_mentions()
+    except Exception as err:
+        LOGGER.log_error("Failed to get mentions: %s" % err)
+        return
+    
     for mention in reversed(mentions):
         print(str(mention.id) + ': ' + mention.full_text)
-        set_last_seen_id(mention)
-
-        if BOT_NAME in mention.full_text:
-            isbns = ISBNFinder.find_isbns(mention.full_text)
-            if not isbns:
-                # fetch tweet's parent (TODO: or siblings) & check for isbns
-                mention = get_parent_tweet_of(mention)
+        # print(json.dumps(mention._json, indent=2))
+
+        try:
+            set_last_seen_id(mention)
+        except Exception as err:
+            LOGGER.log_error("Failed to set last seen id: %s" % err)
+            continue 
+        
+        if BOT_NAME in mention.full_text: # I think I can remove this line. get_latest_mentions should handle
```

Comment:
This may be redundant @robertIanClarkson mentions because we're assuming a mention will always contain the BOT_NAME, e.g. even if it's not in the raw text in the UI the "replying to `@foo`" is encoded in the message we're receiving.